gateway-service-impl/src/main/java/org/hypertrace/gateway/service/baseline/BaselineServiceImpl.java

```diff
@@ -120,12 +128,16 @@ public BaselineEntitiesResponse getBaselineForEntities(
       Period timeSeriesPeriod =
           getTimeSeriesPeriod(originalRequest.getBaselineMetricSeriesRequestList());
       long periodSecs = getPeriodInSecs(timeSeriesPeriod);
+      long alignedStartTime =
+          QueryExpressionUtil.alignToPeriodBoundary(
+              originalRequest.getStartTimeMillis(), periodSecs, true);
+      long alignedEndTime =
+          QueryExpressionUtil.alignToPeriodBoundary(
+              originalRequest.getEndTimeMillis(), periodSecs, false);
       List<TimeAggregation> timeAggregations =
           getTimeAggregationsForTimeSeriesExpr(originalRequest);
-      long seriesStartTime =
-          getUpdatedStartTime(
-              originalRequest.getStartTimeMillis(), originalRequest.getEndTimeMillis());
-      long seriesEndTime = originalRequest.getStartTimeMillis();
+      long seriesStartTime = getUpdatedStartTime(alignedStartTime, alignedEndTime);
+      long seriesEndTime = alignedStartTime;
```

Comment:
should this be equal to alignedEndTime?

Reply:
No, for baseline calculation we are not including current data.